tests/tensor/test_subtensor.py

```diff
@@ -105,13 +105,45 @@ def test_as_index_literal():
 
 
 class TestGetCanonicalFormSlice:
+    def test_none_constant(self):
+        slc, rev = get_canonical_form_slice(NoneConst, 5)
+        assert isinstance(slc, slice)
+        assert slc.start == 0
+        assert slc.stop == 5
+        assert slc.step == 1
+        assert rev == 1
+
     def test_scalar_constant(self):
         a = as_scalar(0)
         length = lscalar()
         res = get_canonical_form_slice(a, length)
         assert res[0].owner.op == ptb.switch
         assert res[1] == 1
 
+    def test_tensor_constant(self):
+        a = as_tensor(0)
+        length = lscalar()
+        res = get_canonical_form_slice(a, length)
+        assert res[0].owner.op == ptb.switch
+        assert res[1] == 1
+
+    def test_symbolic_slice(self):
+        idx = make_slice(slice(3, 7, 2))
```

Comment:
This is the part I was confused, and I thought the function was meant to handle originally, but I don't think it was after more careful reading. If it was mean't to handle it, then it should also handle `make_slice`, with symbolic inputs.

Reply:
No it can handle variable and constant symbolic slices too 😎